### ISSUE TITLE
[BUG] pdk errors out on missing aws-cognito-identitypool-alpha dependency

### ISSUE THREAD

**Issue by vincentgna** [aws/aws-pdk#580]
### Describe the bug

In an existing monorepo running

```sh
pdk new monorepo-ts --no-git --licensed false
```
creates projen project with missing peer dependencies causing pdk to fail on subsequent runs until missing dev dependency is added manually.

### Expected Behavior

no errors when running `pdk` or `yarn projen` after using the monorepo-ts Projen template

### Current Behavior

during `pdk new monorepo-ts` following warnings are printed:

```sh
warning " > @aws/pdk@0.22.6" has unmet peer dependency "@aws-cdk/aws-cognito-identitypool-alpha@^2.93.0-alpha.0".
warning "@aws/pdk > @pnpm/reviewing.dependencies-hierarchy > @pnpm/lockfile-file@8.1.2" has unmet peer dependency "@pnpm/logger@^5.0.0".
```

Running `yarn projen` (or `pdk`) fails with error message:

```sh
yarn projen
yarn run v1.22.19
$ /Users/<redacted>/node_modules/.bin/projen
👾 default | ts-node --project tsconfig.dev.json .projenrc.ts
Error: Cannot find module '@aws-cdk/aws-cognito-identitypool-alpha'
Require stack:
- /Users/<redacted>/node_modules/@aws/pdk/identity/user-identity.js
- /Users/<redacted>/node_modules/@aws/pdk/identity/index.js
- /Users/<redacted>/node_modules/@aws/pdk/index.js
- /Users/<redacted>/.projenrc.ts
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1039:15)
    at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (/Users/<redacted>/node_modules/@cspotcode/source-map-support/source-map-support.js:811:30)
    at Function.Module._load (node:internal/modules/cjs/loader:885:27)
    at Module.require (node:internal/modules/cjs/loader:1105:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/Users/<redacted>/node_modules/@aws/pdk/identity/user-identity.ts:3:1)
    at Module._compile (node:internal/modules/cjs/loader:1218:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1272:10)
    at Object.require.extensions.<computed> [as .js] (/Users/<redacted>/node_modules/ts-node/src/index.ts:1608:43)
    at Module.load (node:internal/modules/cjs/loader:1081:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/<redacted>/node_modules/@aws/pdk/identity/user-identity.js',
    '/Users/<redacted>/node_modules/@aws/pdk/identity/index.js',
    '/Users/<redacted>/node_modules/@aws/pdk/index.js',
    '/Users/<redacted>/.projenrc.ts'
```

### Reproduction Steps

Init a test repo as follows (optional, this was roughly my existing repo setup):

```sh
#!/bin/bash

mkdir pdk-test
pushd pdk-test
{
  touch README.md
  mkdir -p states/{foo,bar,baz}
  touch states/{foo,bar,baz}/main.tf
  mkdir -p modules/{account,network,database}
  touch modules/{account,network,database}/main.tf
  mkdir -p .github/workflows
  touch .github/workflows/build.yml
  echo "v18" >.nvmrc
  nvm use
  echo 'node_modules' >.gitignore
  git init
  git add -A
  git commit -m"Initial commit"
  npx @aws/pdk new monorepo-ts --no-git --licensed false
}
popd
```

after running this script, following steps reproduce the failure

```sh
cd pdk-test
nvm use
yarn projen
```

### Possible Solution

editing `projenrc.ts` to add `aws-cognito-identitypool-alpha` to the `devDeps` array and running `yarn add -W @aws-cdk/aws-cognito-identitypool-alpha` before running `yarn projen` again gets you back into working state.

### Additional Information/Context

```sh
node --version
v18.13.0
```

### PDK version used

0.22.6

### What languages are you seeing this issue on?

Typescript

### Environment details (OS name and version, etc.)

Darwin Kernel Version 21.6.0 arm64

**Comment by agdimech** [comment]
Hi Vincent - sorry for the delayed response.

This is due to the fact that yarn is the only package manager that actually does not install peer dependencies for you whereas `pnpm` or `npm` will actually install them for you. 

I suggest that you switch your package manager to `pnpm` which will handle the auto installation for you and will give you the added benefit of faster install times.

I will keep this ticket open for a week or so before closing.

**Comment by cogwirrel** [comment]
While `pnpm` is definitely the recommended package manager given the speed boost it provides, we've fixed this in `0.22.27` by adding it as a dependency of the TS monorepo by default so you shouldn't need to add it explicitly for yarn.
